fuel-block-producer/src/block_producer/transaction_selector.rs

```diff
@@ -0,0 +1,172 @@
+use std::cmp::Reverse;
+
+use crate::Config;
+use fuel_core_interfaces::common::{
+    fuel_tx::CheckedTransaction,
+    fuel_types::Word,
+};
+
+pub fn select_transactions(
+    mut includable_txs: Vec<CheckedTransaction>,
+    config: &Config,
+) -> Vec<CheckedTransaction> {
+    // Select all txs that fit into the block, preferring ones with higher gas price.
+    //
+    // Future improvements to this algorithm may take into account the parallel nature of
+    // transactions to maximize throughput.
+    let mut used_block_space: Word = 0;
+
+    // Sort transactions by gas price, highest first
+    includable_txs.sort_by_key(|a| Reverse(a.transaction().gas_price()));
+
+    // Pick as many transactions as we can fit into the block (greedy)
+    includable_txs
+        .into_iter()
+        .filter(|tx| {
+            let tx_block_space = tx.max_fee();
+            if let Some(new_used_space) = used_block_space.checked_add(tx_block_space) {
+                if new_used_space <= config.max_gas_per_block {
+                    used_block_space = new_used_space;
+                    true
+                } else {
+                    false
+                }
+            } else {
+                false
+            }
+        })
+        .collect()
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
I'd rather see this kind of integration testing after the block producer is integrated into fuel-core, and for the tests to co-exist with the other integ tests in the standard `fuel-tests` package.

Keeping the dev-dependencies light in these modules allows for rapid iteration & testing of logical changes in a localized way. This is one of the key benefits of a P&A approach, as these "domain modules" can be iterated on and be well-tested independently of each other.

I'm ok with keeping this here temporarily until we finish the integration with fuel-core, but it should be relocated & refactored as part of the integration task.